### ISSUE TITLE
Use `PRGA` in double factorization `ProgRotGateArray`

### ISSUE THREAD

**Issue by anurudhp** [quantumlib/Qualtran#1319]
https://github.com/quantumlib/Qualtran/blob/683472a25a1bafb64fd5e5c58f79405737a4afd4/qualtran/bloqs/chemistry/df/select_bloq.py#L30

**Comment by fdmalone** [comment]
I don't think this is actually equivalent. PRGA acts on a single target but this one needs to act on multiple targets. Ideally would be unified with the THC bloq (xref: #573).

**Comment by fdmalone** [comment]
Perhaps an intermediate step would be to not name this ProgRotGateArray and use something like #579 which may be a lot closer to having a real implementation given all the work on phase gradient rotations etc.
